cmd/create/cluster/cmd.go

```diff
@@ -2064,18 +2061,6 @@ func run(cmd *cobra.Command, _ []string) {
 			r.Reporter.Errorf("Expected a valid private-link value: %s", err)
 			os.Exit(1)
 		}
-		if isHostedCP {
-			privateIngress, err = interactive.GetBool(interactive.Input{
-				Question: "Private ingress",
-				Help: fmt.Sprintf("Private ingress allows you to change the default level of ingress " +
-					"visibility"),
-				Default: privateIngress,
-			})
-			if err != nil {
-				r.Reporter.Errorf("Expected a valid private ingress value: %s", err)
-				os.Exit(1)
-			}
-		}
 	} else if (privateLink || (isSTS && private)) && !fedramp.Enabled() && !isPrivateHostedCP {
```

Comment:
Private API prompt should probably go above this block and the `isPrivateHostedCP` variable declaration so we get the expected warning that privateLink will default to true because the cluster is private HCP - then you should be able to remove the new code that sets privateLink on lines 2083-2087